modern/base/Dockerfile

```diff
@@ -0,0 +1,93 @@
+FROM ubuntu:xenial
+
+LABEL maintainer="Conan.io <info@conan.io>"
+
+ENV PYENV_ROOT=/opt/pyenv \
+    PATH=/opt/pyenv/shims:${PATH}
+
+ARG CMAKE_VERSION CMAKE_VERSION_FULL PYTHON_VERSION
+
+WORKDIR /root
+
+RUN apt-get -qq update \
+    && apt-get -qq install -y --no-install-recommends --no-install-suggests -o=Dpkg::Use-Pty=0 \
+       sudo \
+       binutils \
+       wget \
+       git \
+       g++-multilib \
+       libgmp-dev \
+       libmpfr-dev \
+       libmpc-dev \
+       nasm \
+       dh-autoreconf \
+       libffi-dev \
+       libssl-dev \
+       pkg-config \
+       subversion \
+       zlib1g-dev \
+       libbz2-dev \
+       libsqlite3-dev \
+       libreadline-dev \
+       xz-utils \
+       curl \
+       libncurses5-dev \
+       libncursesw5-dev \
+       liblzma-dev \
+       ca-certificates \
+       autoconf-archive \
+    && rm -rf /var/lib/apt/lists/* \
+    && wget -q --no-check-certificate https://cmake.org/files/v${CMAKE_VERSION}/cmake-${CMAKE_VERSION_FULL}-Linux-x86_64.tar.gz \
+    && tar -xzf cmake-${CMAKE_VERSION_FULL}-Linux-x86_64.tar.gz \
+       --exclude=bin/cmake-gui \
+       --exclude=doc/cmake \
+       --exclude=share/cmake-${CMAKE_VERSION}/Help \
+       --exclude=share/vim \
+       --exclude=share/vim \
+    && cp -fR cmake-${CMAKE_VERSION_FULL}-Linux-x86_64/* /usr \
+    && rm -rf /root/cmake-${CMAKE_VERSION_FULL}-Linux-x86_64 \
+    && rm /root/cmake-${CMAKE_VERSION_FULL}-Linux-x86_64.tar.gz \
+    && curl -fL https://getcli.jfrog.io | sh \
+    && mv jfrog /usr/local/bin/jfrog \
+    && chmod +x /usr/local/bin/jfrog \
+    && groupadd 1001 -g 1001 \
+    && groupadd 1000 -g 1000 \
+    && groupadd 2000 -g 2000 \
+    && groupadd 999 -g 999 \
+    && useradd -ms /bin/bash conan -g 1001 -G 1000,2000,999 \
+    && printf "conan:conan" | chpasswd \
+    && adduser conan sudo \
```

Comment:
we already know conan user is problematic for CI servies: #136
should we use it for new images? why it might be required?

Reply:
It's a problem related to Azure, but we I prefer following security recommendations than allowing Azure. If necessary, we can move to Jenkins.

There some good blog posts about it:
cloud.google.com/architecture/best-practices-for-building-containers
snyk.io/blog/10-docker-image-security-best-practices